torch/testing/_internal/common_distributed.py

```diff
@@ -842,6 +852,66 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for DDP based Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
```

Comment:
i wonder if we really need a new class, or we should just update the behavior of MultiProcessTestCase in a safe way?

It seems like all of these changes are designed to be 'safe' but the current PR doesn't test this.  Modifying the existing class would force running it on all the existing tests and make sure.

Reply:
i think its fine to make all the extensive changes at once, given they are exercised by CI.  its actually riskier to land a new thing that isn't well tested in case it doesn't really work as advertised or in case we only half-migrate to it.

also cc @fegin who has been thinking more about test regfactors

Reply:
i think its fine to make all the extensive changes at once, given they are exercised by CI.  its actually riskier to land a new thing that isn't well tested in case it doesn't really work as advertised or in case we only half-migrate to it.

also cc @fegin who has been thinking more about test base refactors.  We should probably decide if we want to adopt `DistributedTestBase` more widely or build more on top of MPTC.

Reply:
Thanks @wconstab  , will check how much is the impact of adapting the change in all files. 

But in general there are lot of different ways this is getting implemented  

For eg: class ```MultiProcContinousTest ```  introduced with https://github.com/pytorch/pytorch/pull/125648 implements  a mechanism that do not use MPTC but devices related nuances are passed to the individual test file for implementation

Whereas class ```DTensorTestBase```  in torch/testing/_internal/distributed/_tensor/common_dtensor.py , inherits MPTC but tries put the devices related nuances in the common code.

and there are a few more

hence, the suggestion of staggered migration to a common base class that takes into account all the challenges module by module. Since Gaudi/XPU wants to use  distributed test content for validation, we can take up this migration effort.

Reply:
I guess our consideration is the number of TestCases to maintain. 
What tests are you interested in today? `test_c10d_...` ? If so, it seems we can integrate the changes into MPTC first. Then gradually to other TCs.